### PR TITLE
Add/provide jetpack version to blazepress widget

### PR DESCRIPTION
**Pull request by timur987**
Related to SELFSERVE-815 (there you will find links to connected PRs in Jetpack and A8C-DSP).
Related PR in Jetpack (https://github.com/Automattic/jetpack/pull/33823)

## Proposed Changes
- Remove usage of an outdated feature 'promote-post/widget-i2' (since we switched to the new version recently and don't plan to move back),
- Populate Jetpack's version, provided by BlazePress' App from Jetpack in `intial_state.sites['blogID'].options.jetpack_version` to BlazePress widget.

### Preparation

- Run DSP server `yarn dsp`
- Run DSP widget (for WPCOM!): `yarn widget:dev:wpcom`
- Adjust `apps/blaze-dashboard/src/load-config.js` by adding:
```js
productionConfig.dsp_widget_js_src = 'http://localhost:3004/widget.js';
```

- Build Blaze App for use in Jetpack (from `apps/blaze-dashboard`):
```sh
BLAZE_DASHBOARD_PACKAGE_PATH=~/sources/jetpack/projects/packages/blaze yarn dev
```

- Build package in Jetpack:
```sh
jetpack install plugins/crm plugins/debug-helper plugins/jetpack plugins/migration \
  plugins/mu-wpcom-plugin plugins/protect plugins/search plugins/social \
  plugins/starter-plugin plugins/super-cache plugins/vaultpress plugins/videopress \
&& jetpack install -r \
&& pnpm jetpack build packages/blaze
```

- Run Jetpack: `jetpack docker up -d && jetpack docker jt-up`

### Actual Testing
- Open your site via Jurassic Tube,
- Open dev tools,
- Go Tools > Advertising,
- Open the " Sources" tab within Dev Tools and find minified sources for BlazePress App,
- There, find "showDSP" function and put a breakpoint at the first line of the function,
- Click "Promote" and make sure that you can see Jetpack version in Dev Tools like on the sample below:
Screenshot 2023-10-30 at 12 04 40

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/state/ui/selectors/get-jetpack-version.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { getJetpackVersion as getVersion } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Returns the site object for the currently selected site.
+ */
+export default function getJetpackVersion( state: AppState ): string | undefined {
```
Comment (sbarbosa) [inline client/state/ui/selectors/get-jetpack-version.ts]: Added a comment in the Jetpack PR: https://github.com/Automattic/jetpack/pull/33823

I think we can make this PR shorter if we move the location of the jetpack_version parameter in the blaze module. That way, the value will get initiated like the rest of the site options, and we won't need to add anything extra to the Blaze Dashboard boot.

**Review by sbarbosa — COMMENTED** [review]
The changes worked OK. I added a comment in the code. I think we can make this change without adding an additional selector to the Calypso store.

File: client/state/ui/selectors/get-jetpack-version.ts (hunk 1 above)
Comment (timur987) [reply client/state/ui/selectors/get-jetpack-version.ts]: Thanks @sbarbosa! Your suggestion is gold 🥇. 
Cleaned up everything in cc6b0d26b7e6958eb29be847a8b735db34835926 and 61a2be5756e09d5bf18a1057e6020f6ca897546b.

**Review by therocket-gr — APPROVED** [review]
tested and it works ok
code changes LGTM but i see that the PR is in draft mode. shall i check it again when it is ready for a review?

**Comment by timur987** [comment]
Thanks, @therocket-gr!

The PR was in Draft mode because I have not finished describing the PR. 
While describing I'll retest the change and figure out is there something to fix,
and will let you know if re-testing is needed.

**Comment by timur987** [comment]
Small update, I've update testing instructions, @therocket-gr. You can just take a look on them and just confirm that while testing you also saw Jetpack's version. On my side, I retested it again and can confirm it works as expected.

**Review by therocket-gr — APPROVED** [review]
tested again and i confirm the jetpack_version exists
Screenshot 2023-10-30 at 16 13 43

File: client/lib/promote-post/index.ts [hunk 2]
```diff
@@ -62,6 +62,7 @@ declare global {
 					apiRoot: string;
 					headerNonce: string;
 				};
+				jetpackVersion?: string;
 				isV2?: boolean;
```
Comment (therocket-gr) [inline client/lib/promote-post/index.ts]: shouldnt we get rid of the line 
isV2 ?

Comment (timur987) [reply client/lib/promote-post/index.ts]: Oh, I definitely missed it. Thanks to you, @therocket-gr, it was removed in e5b90879c6d6f052fced5c0d2f2cc0333a913883.

**Pull request merged**